### PR TITLE
Add changelog entry for removal of proprietary sphinx directives.

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

Changelog for #11295

### CODE REVIEW

File: doc/api/next_api_changes/2018-05-26-TH.rst [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Removed proprietary sphinx directives
```
Comment (QuLogic) [inline doc/api/next_api_changes/2018-05-26-TH.rst]: I don't know if I'd call these proprietary; the word that I'd like to suggest escapes me at the moment, though.

Comment (dstansby) [reply doc/api/next_api_changes/2018-05-26-TH.rst]: "internally defined"? I agree that I wouldn't call them proprietary.

Comment (jklymak) [reply doc/api/next_api_changes/2018-05-26-TH.rst]: “In-house”?

**Comment by timhoffm** [comment]
Too late to change the suggested wording. "internally defined" would also be fine by me. Feel free to make a separate PR if you want to.

**Pull request merged**